Birdee/Parser.cpp

```diff
@@ -1206,14 +1208,25 @@ std::unique_ptr<FunctionAST> ParseDeclareFunction(ClassAST* cls)
 
 //is_field: true for FieldDef, false for MemberFunctionDef
 //index: the index within fields/funcdef
+static const string INTERNAL_ANNO_VIRTUAL = "virtual";
+static const string INTERNAL_ANNO_PURE_VIRTUAL = "pure_virtual";
```

Comment:
"pure_virtual" maybe => "abstract"? Which one do you prefer?

Reply:
'pure_virtual' describes method, and ‘abstract’ describes class?

Reply:
Java has abstract key word on method. It is shorter and does not have an underscore

Reply:
An example of Java. We don't need an "abstract" before "class" since we can automatically mark the class as abstract. 

```java
abstract class Base { 
    Base() { System.out.println("Base Constructor Called"); } 
    abstract void fun(); 
} 
```

BTW, I think using "abstract" or "pure\_virtual" as an annotation may not be good, as they should be a keyword, not an annotation. In my understanding, "Annotations" are additional info on the AST - If you remove the annotations, the program still compiles. In your case, "pure\_virtual" serves as a token to control the parsing process of the program. It should be a keyword , like in Java.